src/realm-settings/PartialImport.tsx

```diff
@@ -278,107 +285,258 @@ export const PartialImportDialog = (props: PartialImportProps) => {
-                  {realmSelectOptions()}
+                  {collisionOptions()}
                 </Select>
               </StackItem>
-            )}
-            <StackItem>
-              <Text>{t("chooseResources")}:</Text>
-              <DataList aria-label="Resources to import" isCompact>
-                {targetHasResource("users") &&
-                  resourceDataListItem("users", "users")}
-                {targetHasResource("groups") &&
-                  resourceDataListItem("groups", "groups")}
-                {targetHasResource("clients") &&
-                  resourceDataListItem("clients", "clients")}
-                {targetHasResource("identityProviders") &&
-                  resourceDataListItem(
-                    "identityProviders",
-                    "identity providers"
-                  )}
-                {targetHasRealmRoles() &&
-                  resourceDataListItem("realmRoles", "realm roles")}
-                {targetHasClientRoles() &&
-                  resourceDataListItem("clientRoles", "client roles")}
-              </DataList>
-            </StackItem>
-            <StackItem>
-              <Text>{t("selectIfResourceExists")}:</Text>
-              <Select
-                isOpen={isCollisionSelectOpen}
-                direction="up"
-                onToggle={() => {
-                  setIsCollisionSelectOpen(!isCollisionSelectOpen);
-                }}
-                onSelect={handleCollisionSelect}
-                placeholderText={t(collisionOption)}
-              >
-                {collisionOptions()}
-              </Select>
-            </StackItem>
-          </>
-        )}
-      </Stack>
-    </Modal>
+            </>
+          )}
+        </Stack>
+      </Modal>
+    );
+  };
+
+  const importCompleteMessage = () => {
+    return t("importSuccess", {
+      added: importResponse?.added,
+      overwritten: importResponse?.overwritten,
+      skipped: importResponse?.skipped,
+    });
+  };
+
+  const loader = async (first?: number, max?: number) => {
+    let last = first! + max!;
+    const length = importResponse?.results.length;
+    if (last > length!) last = length!;
+
+    if (importResponse?.results)
+      return importResponse.results.slice(first, last);
+
+    // Empty state can't happen as you will get back a result record
+    // for each record you submit. This is here to make Typescript happy.
+    return [];
+  };
+
+  const ActionLabel = (importRecord: { action: string }) => {
+    let text = t("added");
+    let color: "green" | "orange" | "purple" = "green";
+
+    if (importRecord.action === "SKIPPED") {
+      text = t("skipped");
+      color = "orange";
+    }
+
+    if (importRecord.action === "OVERWRITTEN") {
+      text = t("overwritten");
+      color = "purple";
+    }
+
+    return <Label color={color}>{text}</Label>;
+  };
+
+  const TypeRenderer = (importRecord: { resourceType: string }) => {
+    const typeMap = new Map<string, string>([
```

Comment:
There is no need to provide a generic type for this make as it can be inferred from the value passed into the constructor.
```suggestion
    const typeMap = new Map([
```